src/config.rkt

```diff
@@ -131,8 +132,23 @@
 
 (define resetters '())
 
-(define (register-reset fn #:priority [priority 0])
+(define (register-reset! fn #:priority [priority 0])
   (set! resetters (cons (cons priority fn) resetters)))
 
+;; Defines a resetter as a parameter
+;; An initialize and reset function must be provided
+;; A finializer may be optionally specified
+(define-syntax define-resetter
```

Comment:
Also IIUC every caller actually has the same `init-fn` and `reset-fn`. And the `finalizer-fn` is just there to deal with GC stuff. It's OK not to simplify this, but it does seem like we could avoid a lot of the ugliness here by just figuring out how to GC the FFIRules.

Reply:
Yep. Maybe if I have some time in the next week I can look again. Probably will involve the GC not moving things